src/Heading.js

```diff
@@ -19,5 +28,40 @@ Heading.propTypes = {
   ...COMMON.propTypes,
   ...TYPOGRAPHY.propTypes
 }
+//
+
+const HeadingInternal = styled.h2`
+  font-weight: ${get('fontWeights.bold')};
+  margin: 0;
+  ${TYPOGRAPHY} ${COMMON};
+`
+
+HeadingInternal.defaultProps = {
+  theme
+}
+
+HeadingInternal.propTypes = {
+  theme: PropTypes.object,
+  ...COMMON.propTypes,
+  ...TYPOGRAPHY.propTypes
+}
+
+function withHeading(level) {
+  const WithHeading = props => (
+    <HeadingInternal
+      as={level}
```

Comment:
Curious how this behaves if someone does `<H1 as={SomeStyledComponent} ...`

Reply:
Good point 🤔 Since it's such a simple component I'm not sure if it's worth the pain to allow people to pass in a styled component to the `as` prop. If folks really need to add some styles to the headings, they could do: 

const CustomHeading = styled(H1)`
 // custom styles
`

I think allowing the `as` prop on this component would open it up for folks to change the heading level to to get a different font size like so:
```
<H1 as='h2'>Heading</H1>
```

which we want to discourage in favor of this (to encourage proper semantics/a11y:

```
<H2 fontSize='headingSizes.h1' />
```

People could even go as far as:

```
<H1 as='p'>HEADING!</H1>
```
which we really wouldn't want.

Reply:
That sounds good to me, although the TypeScript and propTypes will still list `as` as a valid prop